config.json

```diff
@@ -9,6 +9,9 @@
         "datapipelines": {
             "active": true
         },
+        "datasets_base": {
```

Comment:
I think ideally we would have `dataset_base` be automatically activated if any of the child dataset modules (i.e. `s3_datasets`) is active and not have to expose it here on the config.json

Configuring the `dataset_base` module as active or not does not really mean anything as it is really the child dataset modules that are meaningful to activate or not

I am not sure how easy it is to do such a thing or if it requires a big shift in the loader logic but if an easy fix

Reply:
I saw your comment that it is required here because of how we load ImportModes of CDK and CDK_CLI_EXTENSION - but can you explain why that is / is there an easy way to resolve?

Reply:
First of all, `datasets_base` will be added in the config.json at some point (check the full design steps in https://github.com/data-dot-all/dataall/issues/1123) because some parameters are relevant for all datasets no only for s3_datasets or redshift_datasets (e.g. confidentiality). That is why I did not give it more thought.

But if you are curious, the issue is a bit hidden, the `loader` has some methods to check that it initialized the modules correctly. With the current implementation I think that **a module cannot have a module interface CDK and not have a CDK_CLI_EXTENSION if it is not a config.json module**.  I think the problem is not that much on the loader but on the way we are using it in the cdk proxy. I will have a closer look

Reply:
I think I found the issue as to why there is an error when we remove `datasets_base` from `config.json`:

- When we `load_modules(modes={ImportMode.CDK_CLI_EXTENSION})`
    - We first import all modules `in_config` as part of `def _load_modules(...)`
    - This would theoretically include `s3_datasets` and `dataset_sharing` at this point in time
    - When we import `dataset_sharing` we have the following line in top level of `__init__.py` that subsequently imports `s3_datasets` and `datasets_base` modules
```
from dataall.modules.dataset_sharing.db.share_object_repositories import ShareEnvironmentResource
```

- Then when we `_check_loading_correct(...)` in loader.py
     - it fails on L252 with `ImportError` since `datasets_base` is in `sys.modules.keys()` but is not in `checked_modules_names` (since `datasets_base` is not in config / expected_load)

Previously we never exposed `datasets_base` or `datasets_sharing` as configurable modules in `config.json` so we avoided this problem. Moving the line 
```
from dataall.modules.dataset_sharing.db.share_object_repositories import ShareEnvironmentResource
```

in `backend/dataall/modules/dataset_sharing/__init__.py` to within the the `def __init__(self):` in `class SharingApiModuleInterface()` resolves this issue on loading modules but may need some additional testing to ensure share APIs still working as expected